src/http/interceptor.js

```diff
@@ -0,0 +1,25 @@
+import { REGISTER_ENDPOINT, AUTHENTICATE_ENDPOINT } from "@/http/endpoints";
+
+const responseInterceptor = (response) => response;
+const errorInterceptor = (error) => {
+  if (error.status === 401) {
```

Comment:
I noticed you have an HTTP status code, please use it here
```suggestion
  if (error.status === HTTP_STATUS_CODE.UNAUTHORIZED) {
```